src/main/java/codeu/controller/NewPasswordServlet.java

```diff
@@ -41,17 +41,27 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
         String reTypedNewPassword = request.getParameter("reTypedNewPassword");
         reTypedNewPassword = Jsoup.clean(reTypedNewPassword, Whitelist.none());
         String username =request.getParameter("usernameVerification");
+        //String numbers = "(.*[0-9].*)";
+
 
         if (!userStore.isUserRegistered(username)) {
             request.setAttribute("error", "That username was not found.");
             request.getRequestDispatcher("/WEB-INF/view/newPassword.jsp").forward(request, response);
             return;
         }
+
         if (!newPassword.equals(reTypedNewPassword)) {
             request.setAttribute("error", "Passwords must match.");
             request.getRequestDispatcher("/WEB-INF/view/newPassword.jsp").forward(request, response);
             return;
         }
+
+        if(newPassword.length()<5||newPassword.length()>12 || !newPassword.matches("(.*[a-z].*)")|| !newPassword.matches( "(.*[0-9].*)")){
```

Comment:
OK, but how about some tests ?

Reply:
You mean in the NewPasswordServletTest?

Reply:
Well, yes, this would definitely warrant a test in NewPasswordServletTest